src/commands.ts

```diff
@@ -781,4 +784,136 @@ export async function getVersion() {
     }
     
     this.resume();
+}
+class GiftCard {
+    alias: string;
+    cloudId: string;
+    // balance: number = 0;
+    sk: string;
+    address: string;
+    svg: string;
+    url:string;
+
+    constructor(alias: string, cloudId: string, sk: string, address: string, svg: string, url: string) {
+        // this.balance = 0;
+        this.alias = alias
+        this.cloudId = cloudId;
+        this.sk = sk;
+        this.address = address;
+        this.svg = svg;
+        this.url = url;
+    }
+}
+
+export async function generateGiftCards(prefix: string, quantity: string, cardBalance: string, authToken: string) {
+
+    this.pause();
+    const cloudUrl = process.env.CLOUD_API_ENDPOINT;
+    console.log("cloudUrl = ", cloudUrl)
+    const [total] = await this.account.getShieldedBalances(true);
+    const requiredTotalSum = this.account.humanToShielded(cardBalance) * BigInt(quantity);
+
+    if (requiredTotalSum > total) {
+        this.echo(`total card balance ${requiredTotalSum} exceeds available funds ${this.account.shieldedToHuman(total)}`)
+        return
+    }
```

Comment:
```suggestion
   const oneCardBalance = this.account.humanToShielded(cardBalance)
   const requiredTotalSum = oneCardBalance * BigInt(quantity);

   await this.account.syncState(); // getMaxAvailableTransfer doesn't update state
   const maxAvailableFunds = await this.account.getMaxAvailableTransfer();
   if (requiredTotalSum > maxAvailableFunds) {
       this.echo(`Total card balance ${requiredTotalSum} exceeds available funds ${this.account.shieldedToHuman(maxAvailableFunds)}`);
       return
   }

   const minTransferAmount = await this.account.minTxAmount();
   if (oneCardBalance < minTransferAmount) {
       this.echo(`Single card balance ${requiredTotalSum} less than minimum transfer amount ${this.account.shieldedToHuman(minTransferAmount)}`);
       return
   }
```

Let's add `cardBalance` checking here as well. If we enter a small card amount (less than minimum allowed), the set of cloud accounts will be created but transfer will fail. To decrease probability of the such scenario we should check amount. Moreover we should check available balance more precisely (include fee and suppose multitransfer case).

To make the suggested code compilable you also should add new method in the `account.ts` @ line 347:

```
public async minTxAmount(): Promise<bigint> {
   return await this.zpClient.minTxAmount();
}
```

Reply:
Sorry, I have found na issue here and have updated the suggestion right now, please take in mind

Reply:
fixed by https://github.com/zkBob/zkbob-console/pull/49/commits/c018afe31515741f1d1cf61a0fa08b98c22cbbc5